cgi-bin/LJ/User/Message.pm

```diff
@@ -183,6 +183,13 @@ sub reset_email {
 
     $update_opts ||= { status => 'T' };
     $update_opts->{email} = $newemail;
+
+    # this is no longer done using update_self
+    my $changepass = delete $update_opts->{password};
+    if ( defined $changepass ) {
+        $u->set_password($changepass);
+    }
+
```

Comment:
I just realized that I should clarify - this isn't strictly necessary since `update_self` does eventually do the correct thing, but since I had to check the value of the password option anyway, I thought the logic would be clearer if I explicitly call `set_password` here.

Reply:
Yeah I like the usage of set_password here, good call!